crates/bindings-macro/src/sats.rs

```diff
@@ -205,7 +209,41 @@ pub(crate) fn derive_satstype(ty: &SatsType<'_>) -> TokenStream {
     }
     let (_, typeid_ty_generics, _) = typeid_generics.split_for_impl();
 
+    let impl_plain_enum_extras = if add_impls_for_plain_enum {
+        // These will mostly be empty as lifetime and type parameters must be constrained
+        // but const parameters don't require constraining.
+        let mut generics = ty.generics.clone();
+        add_type_bounds(&mut generics, &quote!(#krate::FilterableValue));
+        let (impl_generics, ty_generics, where_clause) = generics.split_for_impl();
+        // As we don't have access to other `derive` attributes,
+        // we don't know if `Copy` was derived,
+        // so we won't impl for the owned type.
```

Comment:
I think this is probably the wrong default assumption. We're only deriving for plain enums, which almost always will be `Copy`. IMO we should assume that they will be and emit the owned-value derive too, as having it is a much better user experience than not.

Reply:
God, I wish Rust allowed using non-generic `where` bounds on items for trait-bounded conditional compilation.

Reply:
The comment which introduces the redundant lifetime is very compelling. We may want to experiment with that in the future. But for now, my inclination is just to assume that all simple enums will be `Copy` and go from there.